stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt

```diff
@@ -40,7 +40,7 @@ public interface ChannelsViewModel {
     }
 }
 
-public class ChannelsViewModelImpl(
+internal class ChannelsViewModelImpl(
```

Comment:
🤔 Do we need `ChannelsViewModel` interface? Maybe we can easily remove that?

Reply:
I think it's a nice, clean way of defining the contract of the ViewModel, even if we won't have multiple implementations of it. Kind of like an MVP contract interface.

I was actually thinking about introducing similiar interfaces for the other ViewModels 😅

Reply:
I think it depends if want to "bind" our view with a specific contract or we rather give users our view model implementation together with binding and if they want to have some custom - they need to write the whole thing by themselves.
I'm not a big fan of creating interfaces that won't be reused and only defines contract because usually, you end up with exactly one implementation which is called ʻInterfaceIMPL` 🙈 
BTW currently we are using `ChannelsViewModelImpl` inside `ChannelsFragment` so we can't make it internal

Reply:
Oops, I don't know how I haven't looked at the usages, but I just realized this whole change is unfeasible. We need the type passed into the factory to be a `ViewModel` of some sort, and an interface can not do that. Time to remove the interface 😄